src/library.cpp

```diff
@@ -1489,6 +1504,267 @@ void lammps_scatter_atoms_subset(void *ptr, char *name,
 }
 #endif
 
+
+/* ----------------------------------------------------------------------
```

Comment:
for adding new functions to library.cpp, you may want to check out the "progguide" branch, where I am currently reworking the library interface files to have the documentation of the library interface automatically generated from the comments in the source code. you can watch the progress currently at https://docs.lammps/org (scroll to the bottom part of the navigation sidebar).

Reply:
I have pushed a new `library.cpp` to the "scattergather" branch of my fork, where I have put all the fix / compute / property/atom checks into the set of functions
```
void lammps_gather(void *, char *, int, int, void *);
void lammps_gather_concat(void *, char *, int, int, void *);
void lammps_gather_subset(void *, char *, int, int, int, int *, void *);
void lammps_scatter(void *, char *, int, int, void *);
void lammps_scatter_subset(void *, char *, int, int, int, int *, void *);
```
where name = "x" , "f" or other atom properties
        "d_name" or "i_name" for fix property/atom quantities
        "f_fix", "c_compute" for fixes / computes
        will return error if fix/compute doesn't isn't atom-based

These could wrap gather_atoms scatter_atoms etc but also allow more general use. I will look at the documentation format asap

Reply:
If you think this is a better idea than the `gather_fix_* functions` then I will merge the branches

Reply:
I currently cannot comment on that, since I have not reached the scatter/gather library functions in my refactoring in #1970 and thus do not yet have a sense of what is good there or not.

Reply:
For usage of the fix either of the branches are OK, so I am happy with whatever you decide.

Reply:
resolving as changes made